To: vim_dev@googlegroups.com
Subject: Patch 7.3.121
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.121
Problem:    Complicated 'statusline' causes a crash. (Christian Brabandt)
Solution:   Check that the number of items is not too big.
Files:	    src/buffer.c


*** ../vim-7.3.120/src/buffer.c	2011-02-01 21:54:56.000000000 +0100
--- src/buffer.c	2011-02-15 14:00:51.000000000 +0100
***************
*** 3460,3465 ****
--- 3461,3478 ----
      prevchar_isitem = FALSE;
      for (s = usefmt; *s; )
      {
+ 	if (curitem == STL_MAX_ITEM)
+ 	{
+ 	    /* There are too many items.  Add the error code to the statusline
+ 	     * to give the user a hint about what went wrong. */
+ 	    if (p + 6 < out + outlen)
+ 	    {
+ 		mch_memmove(p, " E541", (size_t)5);
+ 		p += 5;
+ 	    }
+ 	    break;
+ 	}
+ 
  	if (*s != NUL && *s != '%')
  	    prevchar_isflag = prevchar_isitem = FALSE;
  
*** ../vim-7.3.120/src/version.c	2011-02-15 11:56:56.000000000 +0100
--- src/version.c	2011-02-15 14:23:39.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     121,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
265. Your reason for not staying in touch with family is that
     they do not have e-mail addresses.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
